### ISSUE TITLE
Create GET endpoints for ClusterRunner logs

### ISSUE THREAD

**Issue by tjlee0909** [box/ClusterRunner#121]
Currently in order to view the logs for: clusterrunner_master, clusterrunner_slave, clusterrunner_deploy, etc, we need to ssh into the particular machine on which those commands ran and open the files in /home/jenkins/.clusterrunner/log/clusterrunner*. 

This is a pretty big pain, and makes it difficult to automate any log aggregation process.

Chad proposed that we create a GET endpoint that would return a JSON array with each log entry being its own JSON array element.

This endpoint would need to accept arguments that at least included:

which log was requested (slave/master/deploy/stop)
how many entries to return (as the number of entries could be huge)
(optional) filter by error, warning, information, notice

**Comment by josephharrington** [comment]
This sounds pretty close to the existing event log. This is currently available via the `/eventlog` endpoint with optional `since_timestamp` query parameter for filtering to recent events. Currently this is available for each host -- e.g., the master logs are available via <master_url>/v1/eventlog, the slave logs are available via <slave_url>/v1/eventlog.

We should definitely do more to enhance this api (add more query params for richer filtering, etc.), but I want to make sure this existing functionality fits the need that this ticket is describing.

**Comment by tjlee0909** [comment]
Ah, I must have misunderstood what Chad was asking for then, I had assumed he wanted the clusterrunner_master.log and clusterrunner_slave.logs.

**Comment by josephharrington** [comment]
@tjlee0909 I don't think you misunderstood -- we should definitely get clarification on the requirement though. We definitely put more info into our human-readable logs (clusterrunner_master.log), but I'm not sure those would translate very well into machine-readable logs (the eventlog).

I think that anything anyone would want to build automation around (dashboards, alerting, etc.) should go into the machine-readable eventlog, otherwise I could foresee a situation where we'd risk making a breaking change merely by changing the wording of some log message string.

**Comment by tjlee0909** [comment]
Yeah, I don't imagine any dashboards (beyond simple ones counting the number of ERROR and WARNING events per day, or something like that) would use this endpoint very much. 

I thought of it more as an endpoint that any clients can use (e.g. Jenkins) to give the users some debugging information (if we implement the build_id query-param) in the console output instead of just the one-liner that we return right now in case of an error.

Or, even just for us, it'd be cool to be able to view the clusterruner_deploy and clusterrunner_master console output logs in the ClusterRunner deploy job in jenkins.

**Comment by tjlee0909** [comment]
I just thought of an example right now--right now, when there is a setup_build failure on the master or slave(s), the error message that the end-user sees from polling the v1/build/<id> endpoint provides just a high-level description of what went wrong. Although we could make those status messages more detailed, it'd be nice to give clients the option of retrieving the full stack trace information with the Python error message.

**Comment by josephharrington** [comment]
Just to note here, the timeline dashboard that we've developed internally at Box and plan to open-source soon uses the eventlog api endpoint exclusively as its datasource.
